### PR TITLE
Facilitator github workflow

### PR DESCRIPTION
**Pull request by aaomidi**
Fixes https://github.com/abetterinternet/prio-server/issues/21

I had to do the build/test coverage creation manually by hand since unfortunately the actions didn't support changing the working directory :(

### CODE REVIEW

File: .github/workflows/facilitator-test.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: facilitator-test
```
Comment (tgeoghegan) [inline .github/workflows/facilitator-test.yml:1]: We already have build, test and clippy in the `build-rust` job in `.github/worksflows/ci-build.yml`. I think the thing to do is add the new coverage target there.

**Review by tgeoghegan — CHANGES_REQUESTED** [review]
Please resolve this with the existing Rust checks in `.github/workflows/ci-build.yml`.

**Comment by aaomidi** [comment]
Do you want me to make another job there or just integrate with `build-rust`? Because I need to use the nightly version of rust for this to work - are we fine changing the build to just use rust nightly?

**Comment by tgeoghegan** [comment]
We should not use the nightly Rust toolchain for the regular CI build, because we want the build and test runs to reflect how we will build production binaries. Also it looks like using the nightly toolchain broke one of the steps in the existing job. I think making a separate job is the right thing, and please include a comment explaining why we need a separate toolchain.

**Comment by tgeoghegan** [comment]
Looking good! Where do we actually get the coverage results though?

**Comment by aaomidi** [comment]
The coverage results will be posted as a comment on the PR and available through the codecov website.https://codecov.io/gh/abetterinternet/prio-server/pull/108

**Review by jsha — APPROVED** [review]

**Review by tgeoghegan — APPROVED** [review]

**Pull request merged**